sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusMessage.cs

```diff
@@ -56,7 +56,7 @@ public ServiceBusMessage(ServiceBusReceivedMessage receivedMessage)
             Label = receivedMessage.Label;
             MessageId = receivedMessage.MessageId;
             PartitionKey = receivedMessage.PartitionKey;
-            Properties = new Dictionary<string, object>(receivedMessage.Properties);
+            Properties = new Dictionary<string, object>(receivedMessage.SentMessage.Properties);
```

Comment:
You should be able to use the read-only collection here, since nothing in the constructor is modifying it.   Any reason to add `SendMessage` in besides avoiding the R/O?

Reply:
Nope, I get this error:
Properties = new Dictionary<string, object>(receivedMessage.Properties);

Argument 1: cannot convert from 'System.Collections.Generic.IReadOnlyDictionary<string, object>' to 'System.Collections.Generic.IDictionary<string, object>'	Azure.Messaging.ServiceBus (netstandard2.0)	C:\Users\jolov\source\repos\Azure\azure-sdk-for-net\sdk\servicebus\Azure.Messaging.ServiceBus\src\ServiceBusMessage.cs	59	Active

Reply:
Not that I find the addition significant in any way, just seems to deviate from how you're pulling the rest in.

Reply:
Fair enough.   I didn't account for it being recognized for the `IDictionary` overload rather than the `IEnumerable<KeyValuePair>` one.   Since you'd have to cast to get there, I think your current version is much more clear.

Reply:
Yeah, ironically this worked without using the SentMessage when using the ImmutableDictionary.